hoodie-client/src/main/java/com/uber/hoodie/index/hbase/HBaseIndex.java

```diff
@@ -99,10 +103,32 @@ public HBaseIndex(HoodieWriteConfig config) {
   }
 
   private void init(HoodieWriteConfig config) {
-    multiPutBatchSize = config.getHbaseIndexGetBatchSize();
-    qpsFraction = config.getHbaseIndexQPSFraction();
-    maxQpsPerRegionServer = config.getHbaseIndexMaxQPSPerRegionServer();
-    putBatchSizeCalculator = new HbasePutBatchSizeCalculator();
+    this.multiPutBatchSize = config.getHbaseIndexGetBatchSize();
+    this.qpsFraction = config.getHbaseIndexQPSFraction();
+    this.maxQpsPerRegionServer = config.getHbaseIndexMaxQPSPerRegionServer();
+    this.putBatchSizeCalculator = new HbasePutBatchSizeCalculator();
+  }
+
+  public HBaseIndexQPSResourceAllocator createQPSResourceAllocator(HoodieWriteConfig config) {
```

Comment:
Make this private ?

Reply:
`putBatchSizeCalculator` is already private? 
Did you mean the method createQPSResourceAllocator() ?

Reply:
Kept public so tests can use this. But I am fine removing the test that is directly calling this method. This is covered in tests that call com.uber.hoodie.index.hbase.HBaseIndex#updateLocation(). Let me know what you think.

Reply:
Yes I mean this method `createQPSResourceAllocator`. Can you add @VisibleforTesting tag to it then ?

Reply:
hey yes, I have added it.